cmd/kaspawallet/daemon/server/address.go

```diff
@@ -39,10 +39,10 @@ func (s *server) ShowAddresses(_ context.Context, request *pb.ShowAddressesReque
 	defer s.lock.Unlock()
 
 	if !s.isSynced() {
-		return nil, errors.New("server is not synced")
+		return nil, errors.Errorf("wallet daemon is not synced yet, %s", s.formatSyncStateReport())
 	}
 
-	addresses := make([]string, 0)
+	addresses := make([]string, 0, s.keysFile.LastUsedExternalIndex())
```

Comment:
Please use `make([]string, s.keysFile.LastUsedExternalIndex())`
Than in line 57 instead of append use `addresses[i] = address.String()`.

It's not very different, but a little safer.

Reply:
It's rather addresses[i-1] = address.String() because this cycle is 1-based